src/providers/ThemeProvider.jsx

```diff
@@ -1,15 +1,21 @@
-import React, { createContext } from 'react';
-import useDarkMode from 'hooks/useDarkMode';
+import React, { createContext, useState } from "react";
+import useDarkMode from "hooks/useDarkMode";
 
 export const ThemeContext = createContext('light');
 
 export default ({ children }) => {
   const [theme, toggleTheme] = useDarkMode();
+  const [scrollPosition, setScrollPosition] = useState(0)
+
+  document.addEventListener('scroll', () => {
+    setScrollPosition(window.scrollY)
+  })
```

Comment:
Can't really think of a solution at right now but my current concern is on every scroll position we set the state to whatever value of `window.scrollY` without condition. I'm wondering if there's a way to disable this whenever we reach a certain threshold. e.g. if > 300, set the value as `true` to change background colour instead of measuring it as a number. This way we we can "probably" improve performance.

Reply:
I have added throttling to improve performance. Now the ```setScrollPosition()``` will be updated after 500ms interval.